crates/turborepo-lib/src/task_hash.rs

```diff
@@ -115,14 +124,74 @@ impl PackageInputsHashes {
                     .unwrap_or_else(|| AnchoredSystemPath::new("").unwrap());
 
                 let scm_telemetry = package_task_event.child();
-                let mut hash_object = match scm.get_package_file_hashes(
-                    repo_root,
-                    package_path,
-                    &task_definition.inputs,
-                    Some(scm_telemetry),
-                ) {
-                    Ok(hash_object) => hash_object,
-                    Err(err) => return Some(Err(err.into())),
+                // Try hashing with the daemon, if we have a connection. If we don't, or if we
+                // timeout or get an error, fallback to local hashing
+                let hash_object = daemon
+                    .as_mut()
+                    .and_then(|daemon| {
+                        let handle = handle.clone();
+                        // We need an async block here because the timeout must be created with an
+                        // active tokio context. Constructing it directly in
+                        // the rayon thread doesn't provide one and will crash at runtime.
+                        handle
+                            .block_on(async {
```

Comment:
I'm a little confused by [the docs](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#panics-2) for this method. It says the method panics "if called within an asynchronous execution context". Does that mean if it's called in an async function?

Reply:
Essentially, yes, if it's running from a task managed by an async runtime, it will panic. In this case, we're running from the rayon threadpool.